core/domain/acl_decorators.py

```diff
@@ -461,21 +455,18 @@ def test_can_manage_collection_publish_status(
             raise base.UserFacingExceptions.PageNotFoundException
 
         if collection_rights.is_published():
-            if role_services.ACTION_UNPUBLISH_PUBLIC_COLLECTION in self.actions:
+            if (role_services.ACTION_UNPUBLISH_ANY_PUBLIC_ACTIVITY in
```

Comment:
So, just to confirm.

I'll break CollectionRightsHandler into three handlers.

One to give collection rights, one to publish the collection and another one to unpublish the collection?